e2e/nomostest/predicates.go

```diff
@@ -169,24 +169,33 @@ func HasExactlyLabelKeys(wantKeys ...string) Predicate {
 	}
 }
 
-// HasExactlyImageName ensures a container has the expected image name
-func HasExactlyImageName(containerName string, expectImageName string) Predicate {
+// HasExactlyImage ensures a container has the expected image.
+func HasExactlyImage(containerName string, expectImageName, expectImageTag, expectImageDigest string) Predicate {
 	return func(o client.Object) error {
 		dep, ok := o.(*appsv1.Deployment)
 		if !ok {
 			return WrongTypeErr(dep, &appsv1.Deployment{})
 		}
 		for _, container := range dep.Spec.Template.Spec.Containers {
 			if containerName == container.Name {
-				if !strings.Contains(container.Image, "/"+expectImageName+":") {
-					return errors.Errorf(" Expected %q container image name is %q, however the actual image is %q", container.Name, expectImageName, container.Image)
+				expectImage := ""
+				if expectImageName != "" {
+					expectImage = "/" + expectImageName
+				}
+				if expectImageTag != "" {
+					expectImage += ":" + expectImageTag
+				}
+				if expectImageDigest != "" {
+					expectImage += "@" + expectImageDigest
+				}
+				if !strings.Contains(container.Image, expectImage) {
+					return errors.Errorf(" Expected %q container image contains %q, however the actual image is %q", container.Name, expectImage, container.Image)
```

Comment:
nit: remove leading space in `" Expected %q ...`

Reply:
good catch :)